### PR TITLE
Markdownification and other fixes to CONTRIBUTING

### PR DESCRIPTION
**Pull request by anoadragon453**
Was reviewing CONTRIBUTING and saw it was quite outdated, and still RST.

So I've converted it to MD and fixed up a bunch of things.

I've left the file as `CONTRIBUTING.rst` for review so that you get a sane diff.

TODO:
 - [ ] Rename file to CONTRIBUTING.md once review is :heavy_check_mark:

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -47,81 +44,85 @@ To run unit tests in a local development environment, you can use:
...
+## Changelog
 
 All changes, even minor ones, need a corresponding changelog / newsfragment
-entry. These are managed by Towncrier
-(https://github.com/hawkowl/towncrier).
+entry. These are managed by [Towncrier](https://github.com/hawkowl/towncrier).
 
-To create a changelog entry, make a new file in the ``changelog.d`` file named
-in the format of ``PRnumber.type``. The type can be one of the following:
+To create a changelog entry, make a new file in the `changelog.d` file named
```
Comment (richvdh) [inline CONTRIBUTING.rst]: while you're here:

```suggestion
To create a changelog entry, make a new file in the `changelog.d` directory named
```

**Comment by anoadragon453** [comment]
> I thought github made a reasonable job of showing diffs on renamed files?

Nah, it's just "all additions in this file, deleted this file".

> I'm going to bet that CONTRIBUTING.rst is linked from lots of places, but I guess we can fix them as we find them.

Ah, yeah I meant to do this actually. Thanks for the reminder, seems it's only really mentioned in the PR template...

**Comment by Half-Shot** [comment]
(We actually reference this in most bridge projects, because copying out the sign off rules in each project seemed silly. We reference `master` though, so will go and update those projects before the next Synapse release)

**Comment by Half-Shot** [comment]
Also https://github.com/matrix-org/sytest/blob/develop/CONTRIBUTING.rst

**Comment by anoadragon453** [comment]
And https://github.com/matrix-org/sygnal/blob/master/CONTRIBUTING.rst :)

**Comment by anoadragon453** [comment]
Happily these projects all link to the master branch though, so they'll need to updated on next Synapse release...

Ideally they'd link to a specific commit instead.

**Comment by richvdh** [comment]
I'm a bit unconvinced that it does make sense for everything to share a CONTRIBUTING. Apart from the DCO bit, I think it's different for each project.

**Comment by Half-Shot** [comment]
>Apart from the DCO bit, I think it's different for each project.

Perhaps that should live in matrix-doc?

**Pull request merged**